### PR TITLE
rename internal use of  behavour to behaviour

### PR DESCRIPTION
**Pull request by matbesancon**
"behavour" seems to be a typo here, I added a deprecation to avoid breaking external code relying on this

### CODE REVIEW

**Comment by simeonschaub** [comment]
This is definitely an internal function and is never called by any external package (https://juliahub.com/ui/RepoSearch?q=_check_add%21%21_behavour), so I don't think we need a deprecation for this.

**Comment by matbesancon** [comment]
Removed the deprecation

**Review by simeonschaub — APPROVED** [review]

**Comment by oxinabox** [comment]
@simeonschaub  you have merge rights,  and @matbesancon  does not AFAIK.
Per [ColPrac](https://github.com/SciML/ColPrac#reviewing-approving-and-merging-prs) you should merge this PR is there is nothing pending.

**Comment by simeonschaub** [comment]
Sorry, I thought best practice was for another committer to look at this and then merge.

**Comment by oxinabox** [comment]
No problem, it's why we write these down.

**Pull request merged**
